### DISCUSSION TITLE
rio.write_crs() deletes the 'grid_mapping' attribute

### DISCUSSION THREAD

**Discussion by hershalpandya** [corteva/rioxarray#795] [Q&A] [answered]
I have inherited a code which has many functions that do not expect CRS to be present in the xarray dataset. So the first thing the function does is to do `ds.rio.write_crs()` and after that other data manipulations happen . Whenever this is done, however, the CRS is written under coordinates group with the name specified in the `ds.attrs['grid_mapping'] `.

[Dockerfile.txt](https://github.com/user-attachments/files/16097325/Dockerfile.txt)
[environmentdotyml.txt](https://github.com/user-attachments/files/16097327/environmentdotyml.txt)

The side effect of this is however that the attribute 'grid_mapping' attribute gets deleted from the `ds`. Is this an intended behavior? Because for my purpose, I need this attribute to stay. Has this been introduced in newer versions of rioxarray ? 

![image](https://github.com/corteva/rioxarray/assets/8216448/190e84f5-3e6b-4d60-b205-94d87859dfc2)

**Comment by snowman2** [comment, marked as answer]
grid_mapping should be in encoding, not attrs. See: in https://github.com/corteva/rioxarray/issues/282

It is recommended to use `rio.grid_mapping` and `rio.write_grid_mapping`.

**Reply by hershalpandya** [reply]
Thanks!
